src/Raven.Server/Documents/ShardedHandlers/ShardedHiLoHandler.cs

```diff
@@ -0,0 +1,88 @@
+// -----------------------------------------------------------------------
+//  <copyright file="ShardedHiLoHandler.cs" company="Hibernating Rhinos LTD">
+//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Raven.Client;
+using Raven.Client.Documents.Identity;
+using Raven.Server.Documents.Handlers;
+using Raven.Server.Documents.ShardedHandlers.ShardedCommands;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.ShardedHandlers
+{
+    public class ShardedHiLoHandler : ShardedRequestHandler
+    {
+        [RavenShardedAction("/databases/*/hilo/next", "GET")]
+        public async Task GetNextHiLo()
+        {
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var tag = GetQueryStringValueAndAssertIfSingleAndNotEmpty("tag");
+                var numOfShards = ShardedContext.RequestExecutors.Length;
```

Comment:
That isn't number of shards, that is the number of _nodes_.
Note that this _can_ change.